File: core/clothing.go
package core

import (
	"github.com/louisevanderlith/husk/hsk"
	"github.com/louisevanderlith/husk/validation"
)

type Clothing struct {
	Code        string `hsk:"size(64)"`
	Brand       hsk.Key
	Description string
	Type        hsk.Key
	Size        string
	Colour      string
	Material    string
	Weight      int //grams
}

func (c Clothing) Valid() error {
	return validation.Struct(c)
}

File: lib/bodies/clothing.dart
import 'package:mango_ui/keys.dart';

class Clothing {
  final String code;
  final Key brand;
  final String description;
  final Key type;
  final String size;
  final String colour;
  final String material;
  final num weight;

  Clothing(this.code, this.brand, this.description, this.type, this.size,
      this.colour, this.material, this.weight);

  Map<String, dynamic> toJson() {
    return {
      "Code": code,
      "Brand": brand,
      "Description": description,
      "Type": type,
      "Size": size,
      "Colour": colour,
      "Material": material,
      "Weight": weight,
    };
  }
}
